pkg/cluster/prepare_test.go

```diff
@@ -0,0 +1,219 @@
+			Expect(err).NotTo(HaveOccurred())
+			p := &cluster.Preparer{
+				PrepConfig: cluster.PrepConfig{
+					Location: tmp,
+					DryRun:   true,
+				},
+				Fetcher: &cluster.Fetcher{
+					Client: client,
+				},
+				Applier: &cluster.Applier{
+					Runner: runner,
+				},
+			}
+			err = p.Prepare()
+			Expect(err).NotTo(HaveOccurred())
+			files, err := ioutil.ReadDir(tmp)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(files) > 0).To(BeTrue())
+			Expect(runner.RunCallCount()).To(Equal(1)) // there should be no call because of dryrun.
+			arg, args := runner.RunArgsForCall(0)
+			Expect(arg).To(Equal("kubectl"))
+			Expect(args).To(Equal([]string{"apply", "-f", tmp, "--dry-run=client"}))
+		})
+	})
+	When("dry-run is not set", func() {
+		It("sets up the environment with everything that profiles needs", func() {
+			runner := &gitfakes.FakeRunner{}
+			content, err := ioutil.ReadFile(filepath.Join("testdata", "manifests.tar.gz"))
+			Expect(err).NotTo(HaveOccurred())
+			client := &http.Client{
+				Transport: &mockTransport{
+					res: &http.Response{
+						StatusCode: http.StatusOK,
+						Body:       ioutil.NopCloser(bytes.NewReader(content)),
+					},
+				},
+			}
+			tmp, err := ioutil.TempDir("", "prepare_02")
+			Expect(err).NotTo(HaveOccurred())
+			p := &cluster.Preparer{
+				PrepConfig: cluster.PrepConfig{
+					BaseURL:  "https://github.com/weaveworks/profiles/releases",
+					Location: tmp,
+				},
+				Fetcher: &cluster.Fetcher{
+					Client: client,
+				},
+				Applier: &cluster.Applier{
+					Runner: runner,
+				},
+			}
+			err = p.Prepare()
+			Expect(err).NotTo(HaveOccurred())
+			files, err := ioutil.ReadDir(tmp)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(files) > 0).To(BeTrue())
+			Expect(runner.RunCallCount()).To(Equal(1))
+			arg, args := runner.RunArgsForCall(0)
+			Expect(arg).To(Equal("kubectl"))
+			Expect(args).To(Equal([]string{"apply", "-f", tmp}))
+		})
+	})
+	When("there is an error downloading the manifests", func() {
+		It("will fail and show a proper error to the user", func() {
+			runner := &gitfakes.FakeRunner{}
+			client := &http.Client{
+				Transport: &mockTransport{
+					res: &http.Response{
+						StatusCode: http.StatusNotFound,
+						Status:     http.StatusText(http.StatusNotFound),
+					},
+				},
+			}
+			tmp, err := ioutil.TempDir("", "prepare_03")
+			Expect(err).NotTo(HaveOccurred())
+			p := &cluster.Preparer{
+				PrepConfig: cluster.PrepConfig{
+					BaseURL:  "https://github.com/weaveworks/profiles/releases",
+					Location: tmp,
+				},
+				Fetcher: &cluster.Fetcher{
+					Client: client,
+				},
+				Applier: &cluster.Applier{
+					Runner: runner,
+				},
+			}
+			err = p.Prepare()
+			Expect(err).To(MatchError("failed to download manifests.tar.gz from https://github.com/weaveworks/profiles/releases/latest/download/manifests.tar.gz, status: Not Found"))
+		})
+	})
+	When("there is an error accessing the output folder", func() {
```

Comment:
speaking of this, do we want to do a stat (or whatever) on this folder immediately so that we fail fast before any client action?